File: ios-native/WordWheel/Views/HelpDialog.swift
import SwiftUI

/// "How to Play" modal. Shown automatically on first launch (gated by
/// `GameStorage.seenHelp`) and via the `?` icon in the TopBar afterwards.
struct HelpDialog: View {
    let onDismiss: () -> Void

    var body: some View {
        ZStack {
            Color.black.opacity(0.78)
                .ignoresSafeArea()
                .onTapGesture(perform: onDismiss)

            VStack(alignment: .center, spacing: 0) {
                Text("How to Play")
                    .font(.system(size: 22, weight: .bold))
                    .foregroundColor(.white)
                    .padding(.top, 24)

                VStack(alignment: .leading, spacing: 6) {
                    HelpRow(glyph: "✋", title: "Drag to spell",
                            detail: "Drag your finger across the wheel letters to build a word. Release to submit.")
                    HelpRow(glyph: "✓", title: "Fill the grid",
                            detail: "Words you find that fit the crossword fill in. Solve them all to clear the level.")
                    HelpRow(glyph: "★", title: "Bonus words",
                            detail: "Extra valid words still earn points — they appear in the row below the wheel.")
                    HelpRow(glyph: "💡", title: "Hints",
                            detail: "Tap the hint button to reveal one grid letter. Earn +1 hint every 10 words you find.")
                    HelpRow(glyph: "🔄", title: "Shuffle",
                            detail: "Tap the centre of the wheel to shuffle the tiles when you're stuck.")
                }
                .padding(.horizontal, 20)
                .padding(.top, 16)

                Button(action: onDismiss) {
                    Text("Got it")
                        .font(.system(size: 16, weight: .bold))
                        .foregroundColor(.white)
                        .padding(.horizontal, 36).padding(.vertical, 12)
                        .background(RoundedRectangle(cornerRadius: 24).fill(GameColors.submitBg))
                }
                .buttonStyle(.plain)
                .padding(.vertical, 22)
            }
            .frame(maxWidth: 420)
            .padding(.horizontal, 12)
            .background(
                RoundedRectangle(cornerRadius: 20).fill(GameColors.completeBg)
            )
            .shadow(color: .black.opacity(0.5), radius: 18)
            .padding(.horizontal, 24)
        }
    }
}

private struct HelpRow: View {
    let glyph: String
    let title: String
    let detail: String

    var body: some View {
        HStack(alignment: .top, spacing: 12) {
            Text(glyph)
                .font(.system(size: 22))
                .frame(width: 32, alignment: .center)
                .padding(.top, 2)
            VStack(alignment: .leading, spacing: 2) {
                Text(title)
                    .font(.system(size: 16, weight: .semibold))
                    .foregroundColor(.white)
                Text(detail)
                    .font(.system(size: 13))
                    .foregroundColor(.white.opacity(0.75))
                    .fixedSize(horizontal: false, vertical: true)
            }
        }
        .padding(.vertical, 4)
    }
}

File: android-native/app/src/main/kotlin/com/wheelword/game/ui/HelpDialog.kt
package com.wheelword.game.ui

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wheelword.game.theme.BalooFamily
import com.wheelword.game.theme.GameColors
import com.wheelword.game.theme.NunitoFamily

/**
 * "How to Play" modal. Shown automatically on first launch (gated by
 * [com.wheelword.game.storage.GameStorage.seenHelp]) and via the `?`
 * icon afterwards.
 */
@Composable
fun HelpDialog(onDismiss: () -> Unit) {
    ModalShell(onDismiss = onDismiss, widthFraction = 0.88f) {
        Text(
            text = "How to Play",
            color = Color.White,
            fontFamily = BalooFamily,
            fontSize = 26.sp,
            fontWeight = FontWeight.ExtraBold,
        )
        Spacer(Modifier.height(14.dp))

        HelpItem(
            glyph = "✋",
            title = "Drag to spell",
            body = "Drag your finger across the wheel letters to build a word. Release to submit.",
        )
        HelpItem(
            glyph = "✓",
            title = "Fill the grid",
            body = "Words you find that fit the crossword fill in. Solve them all to clear the level.",
        )
        HelpItem(
            glyph = "★",
            title = "Bonus words",
            body = "Extra valid words still earn points — they appear in the row below the wheel.",
        )
        HelpItem(
            glyph = "💡",
            title = "Hints",
            body = "Tap the hint button to reveal one grid letter. Earn +1 hint every 10 words you find.",
        )
        HelpItem(
            glyph = "🔄",
            title = "Shuffle",
            body = "Tap the centre of the wheel to shuffle the tiles when you're stuck.",
        )

        Spacer(Modifier.height(20.dp))
        CandyButton(
            text = "Got it",
            onClick = onDismiss,
            color = GameColors.Blue,
            colorDeep = GameColors.BlueDeep,
            fontSize = 17.sp,
            modifier = Modifier.fillMaxWidth(),
            verticalPadding = 11.dp,
        )
    }
}

@Composable
private fun HelpItem(glyph: String, title: String, body: String) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 6.dp),
        verticalAlignment = Alignment.Top,
    ) {
        Text(
            text = glyph,
            fontSize = 22.sp,
            modifier = Modifier
                .width(36.dp)
                .padding(top = 2.dp),
        )
        Column(modifier = Modifier.weight(1f)) {
            Text(
                text = title,
                color = Color.White,
                fontFamily = BalooFamily,
                fontSize = 16.sp,
                fontWeight = FontWeight.Bold,
            )
            Text(
                text = body,
                color = GameColors.InkSoft,
                fontFamily = NunitoFamily,
                fontWeight = FontWeight.SemiBold,
                fontSize = 13.sp,
            )
        }
    }
}
